test/test_timers.py

```diff
@@ -116,16 +141,27 @@ def test_timer_error(container_factory, caplog):
     )
 
 
-def _get_timer_test_service(interval, eager, fcn=None):
-    """Construct a dummy nameko service with a simple timer entrypoint."""
-    fcn = fcn or Mock()
+def test_expected_error_in_worker(container_factory, caplog):
+    """Make sure that expected exceptions are processed correctly."""
+    class ExampleError(Exception):
+        pass
 
     class Service(object):
         name = "service"
 
-        @Timer.decorator(interval, eager)
+        @timer(1, expected_exceptions=(ExampleError,))
         def tick(self):
-            return fcn()
+            raise ExampleError('boom!')
+
+    container = container_factory(Service, {})
 
-    Service.fcn = fcn
-    return Service
+    with entrypoint_hook(container, "tick") as tick:
+        container.start()
+        with pytest.raises(ExampleError):
+            tick()
+    assert not container._died.ready()
```

Comment:
It wouldn't be expected for the container to die even if the error wasn't expected, so this line is misleading. You can probably just drop it.